packages/compass-web/sandbox/stored-connections-history.tsx

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useState, useEffect } from 'react';
+import type { ConnectionInfo } from '@mongodb-js/connection-storage/renderer';
+import { Label } from '@mongodb-js/compass-components';
+import { redactConnectionString } from 'mongodb-connection-string-url';
+import React from 'react';
+import { ConnectionsList } from './connections-list';
+
+function getHistory(): ConnectionInfo[] {
+  try {
+    const b64Str = localStorage.getItem('CONNECTIONS_HISTORY');
+    if (!b64Str) {
+      return [];
+    }
+    const binStr = window.atob(b64Str);
+    const bytes = Uint8Array.from(binStr, (v) => v.codePointAt(0) ?? 0);
+    const str = new TextDecoder().decode(bytes);
+    return JSON.parse(str);
+  } catch (err) {
+    return [];
+  }
+}
+export function saveHistory(history: any) {
+  try {
+    const bytes = new TextEncoder().encode(JSON.stringify(history));
+    const binStr = String.fromCodePoint(...bytes);
+    const b64Str = window.btoa(binStr);
+    localStorage.setItem('CONNECTIONS_HISTORY', b64Str);
+  } catch (err) {
+    // noop
+  }
+}
+export function useConnectionsHistory() {
+  const [connectionsHistory, setConnectionsHistory] = useState(() => {
+    return getHistory();
+  });
+
+  useEffect(() => {
+    saveHistory(connectionsHistory);
+  }, [connectionsHistory]);
+
+  const updateConnectionsHistory = useCallback(
+    (connectionInfo: ConnectionInfo) => {
+      setConnectionsHistory((history) => {
+        const existingIndex = history.findIndex((info) => {
+          return (
+            info.connectionOptions.connectionString ===
+            connectionInfo.connectionOptions.connectionString
+          );
+        });
+        if (existingIndex !== -1) {
+          return history;
+        }
```

Comment:
Using history.some is probably easier to understand if we don't need that specific history item later.

Reply:
Yep, definitely reads easier, updated in https://github.com/mongodb-js/compass/pull/5634/commits/e1ac8a0f5f31bb0b4949389468202002fcb56e47